File: app/models/batch_test.py
"""
批量测试模块数据模型

用于对 Dify Agent 进行批量测试，支持导入 Excel/CSV 文件，
选择 Dify Agent，系统自动执行测试并记录结果。
"""
from tortoise import fields

from .base import BaseModel, TimestampMixin


class BatchTestTask(BaseModel, TimestampMixin):
    """
    批量测试任务表

    存储每次导入的批量测试任务信息
    """
    tenant_id = fields.BigIntField(default=0, description="租户ID", index=True)
    task_name = fields.CharField(max_length=200, description="任务名称", index=True)
    version_no = fields.IntField(default=1, description="版本号，从1开始，每次重新执行+1")
    dify_agent_id = fields.BigIntField(description="Dify Agent ID", index=True)
    file_name = fields.CharField(max_length=500, description="原始文件名")
    file_size = fields.IntField(default=0, description="文件大小(字节)")
    row_count = fields.IntField(default=0, description="数据行数")
    collection_name = fields.CharField(max_length=200, description="SeekDB集合名称（包含版本号）")
    status = fields.IntField(default=0, description="任务状态：0-待执行, 1-执行中, 2-已完成, 3-失败, 4-已停止")
    success_count = fields.IntField(default=0, description="成功数")
    fail_count = fields.IntField(default=0, description="失败数")
    remark = fields.CharField(max_length=1000, default="", description="备注")
    created_by = fields.BigIntField(default=0, description="创建人ID")

    class Meta:
        table = "batch_test_task"

File: frontend/src/api/autofill/batch_test.ts
import request from '@/utils/request'

// 批量测试任务接口

export interface BatchTestTask {
  id: number
  task_name: string
  version_no: number
  dify_agent_id: number
  dify_agent_name: string
  file_name: string
  file_size: number
  row_count: number
  collection_name: string
  status: number
  success_count: number
  fail_count: number
  remark: string
  created_by: number
  created_at: string
  updated_at: string
}

export interface BatchTestTaskQuery {
  page?: number
  page_size?: number
  keyword?: string
}

export interface BatchTestResultQuery {
  page?: number
  page_size?: number
  status?: number
}

export interface BatchTestImportResponse {
  task_id: number
  task_name: string
  version_no: number
  row_count: number
  status: number
  collection_name: string
}

export interface BatchTestFilePreview {
  file_name: string
  file_size: number
  row_count: number
  headers: string[]
  sample_data: Record<string, string>[]
  has_query_column: boolean
}

// 预览文件
export function previewBatchTestFile(file: File) {
  const formData = new FormData()
  formData.append('file', file)
  return request({
    url: '/batch-test/tasks/preview',
    method: 'post',
    data: formData,
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

// 导入文件创建任务
export function importBatchTestFile(
  file: File,
  difyAgentId: number,
  taskName?: string,
  remark?: string
) {
  const formData = new FormData()
  formData.append('file', file)
  formData.append('dify_agent_id', difyAgentId.toString())
  if (taskName) formData.append('task_name', taskName)
  if (remark) formData.append('remark', remark)

  return request({
    url: '/batch-test/tasks/import',
    method: 'post',
    data: formData,
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

// 获取任务列表
export function getBatchTestTaskList(params: BatchTestTaskQuery) {
  return request({
    url: '/batch-test/tasks/list',
    method: 'get',
    params
  })
}

// 获取任务详情
export function getBatchTestTaskDetail(taskId: number) {
  return request({
    url: `/batch-test/tasks/${taskId}`,
    method: 'get'
  })
}

// 停止任务
export function stopBatchTestTask(taskId: number) {
  return request({
    url: `/batch-test/tasks/${taskId}/stop`,
    method: 'post'
  })
}

// 删除任务
export function deleteBatchTestTask(taskId: number) {
  return request({
    url: `/batch-test/tasks/${taskId}`,
    method: 'delete'
  })
}

// 重新执行任务
export function retryBatchTestTask(taskId: number) {
  return request({
    url: `/batch-test/tasks/${taskId}/retry`,
    method: 'post'
  })
}

// 获取测试结果列表
export function getBatchTestResults(taskId: number, params: BatchTestResultQuery) {
  return request({
    url: `/batch-test/tasks/${taskId}/results`,
    method: 'get',
    params
  })
}

// 导出测试结果
export function exportBatchTestResults(taskId: number) {
  return request({
    url: `/batch-test/tasks/${taskId}/export`,
    method: 'get',
    responseType: 'blob'
  })
}

// 获取 Dify Agent 下拉列表
export function getDifyAgentSelect() {
  return request({
    url: '/autofill/dify-agent/select',
    method: 'get'
  })
}
